src/Symfony/Component/Translation/Translator.php

```diff
@@ -241,36 +241,52 @@ public function transChoice($id, $number, array $parameters = array(), $domain =
         }
 
         $id = (string) $id;
-        $catalogue = $this->getCatalogue($locale);
+        $bestLocale = $this->resolveLocale($id, $domain, $locale);
+
+        if (null !== $bestLocale) {
+           return strtr($this->selector->choose($this->getCatalogueInternal($locale)->get($id, $domain), (int) $number, $bestLocale), $parameters);
+        } else {
+           return strtr($this->selector->choose($id, (int) $number, $locale), $parameters);
```

Comment:
Might be an edge case not covered by tests – which locale should the selector apply when the message has no translation, i. e. we're using the `$id` itself?